tests/ui/LHNItemsPresence.tsx

```diff
@@ -357,5 +359,140 @@ describe('SidebarLinksData', () => {
             // Then the empty report should not appear in the sidebar.
             expect(getOptionRows()).toHaveLength(0);
         });
+
+        it('should not display the report marked as hidden', async () => {
+            // When the SidebarLinks are rendered.
+            LHNTestUtils.getDefaultRenderedSidebarLinks();
+            const report: Report = {
+                ...createReport(),
+                participants: {
+                    [TEST_USER_ACCOUNT_ID]: {
+                        notificationPreference: CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN,
+                    },
+                },
+            };
+
+            // And a report marked as hidden is initialized in Onyx
+            await initializeState({
+                [`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`]: report,
+            });
+
+            // Then hidden report should not appear in the sidebar.
+            expect(getOptionRows()).toHaveLength(0);
+        });
```

Comment:
We are actually doind some changes to this flow, can you also add a test (if not present) to make sure the report that has empty notification preference is also not rendered?